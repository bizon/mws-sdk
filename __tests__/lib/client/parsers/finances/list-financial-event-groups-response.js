const parseXml = require('../../../../../lib/client/parsers')
const parseListFinancialEventGroupsResponse = require('../../../../../lib/client/parsers/finances/list-financial-event-groups-response')

describe('lib.client.parsers.finances.list-financial-event-groups-response', () => {
  it('should parse the ListFinancialEventGroupsResponse example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <ListFinancialEventGroupsResponse xmlns="http://mws.amazonservices.com/Finances/2015-05-01">
        <ListFinancialEventGroupsResult>
          <NextToken>2YgYW55IGNhcm5hbCBwbGVhcEXAMPLE</NextToken>
          <FinancialEventGroupList>
            <FinancialEventGroup>
              <FinancialEventGroupId>
                22YgYW55IGNhcm5hbCBwbGVhEXAMPLE
              </FinancialEventGroupId>
              <ProcessingStatus>Closed</ProcessingStatus>
              <FundTransferStatus>Successful</FundTransferStatus>
              <OriginalTotal>
                <CurrencyCode>USD</CurrencyCode>
                <Amount>19.00</Amount>
              </OriginalTotal>
              <ConvertedTotal>
                <CurrencyCode>USD</CurrencyCode>
                <Amount>19.00</Amount>
              </ConvertedTotal>
              <FundTransferDate>2014-09-09T01:30:00.000-06:00</FundTransferDate>
              <TraceId>128311029381HSADJEXAMPLE</TraceId>
              <AccountTail>1212</AccountTail>
              <BeginningBalance>
                <CurrencyCode>USD</CurrencyCode>
                <Amount>0.00</Amount>
              </BeginningBalance>
              <FinancialEventGroupStart>
                2014-09-01T01:30:00.000-06:00
              </FinancialEventGroupStart>
              <FinancialEventGroupEnd>
                2014-09-09T01:30:00.000-06:00
              </FinancialEventGroupEnd>
            </FinancialEventGroup>
          </FinancialEventGroupList>
        </ListFinancialEventGroupsResult>
        <ResponseMetadata>
          <RequestId>1105b931-6f1c-4480-8e97-f3b46EXAMPLE</RequestId>
        </ResponseMetadata>
      </ListFinancialEventGroupsResponse>`
    )

    const res = parseListFinancialEventGroupsResponse(
      '/finances:ListFinancialEventGroupsResponse',
      doc
    )

    expect(res).toMatchSnapshot()
  })

  it('should parse the ListFinancialEventGroupsByNextTokenResponse example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <ListFinancialEventGroupsByNextTokenResponse xmlns="http://mws.amazonservices.com/Finances/2015-05-01">
        <ListFinancialEventGroupsByNextTokenResult>
          <NextToken>2YgYW55IGNhcm5hbCBwbGVhcEXAMPLE</NextToken>
          <FinancialEventGroupList>
            <FinancialEventGroup>
              <FinancialEventGroupId>
                22YgYW55IGNhcm5hbCBwbGVhEXAMPLE
              </FinancialEventGroupId>
              <ProcessingStatus>Closed</ProcessingStatus>
              <FundTransferStatus>Successful</FundTransferStatus>
              <OriginalTotal>
                <CurrencyCode>USD</CurrencyCode>
                <Amount>19.00</Amount>
              </OriginalTotal>
              <ConvertedTotal>
                <CurrencyCode>USD</CurrencyCode>
                <Amount>19.00</Amount>
              </ConvertedTotal>
              <FundTransferDate>2014-09-09T01:30:00.000-06:00</FundTransferDate>
              <TraceId>128311029381HSADJEXAMPLE</TraceId>
              <AccountTail>1212</AccountTail>
              <BeginningBalance>
                <CurrencyCode>USD</CurrencyCode>
                <Amount>0.00</Amount>
              </BeginningBalance>
              <FinancialEventGroupStart>
                2014-09-01T01:30:00.000-06:00
              </FinancialEventGroupStart>
              <FinancialEventGroupEnd>
                2014-09-09T01:30:00.000-06:00
              </FinancialEventGroupEnd>
            </FinancialEventGroup>
          </FinancialEventGroupList>
        </ListFinancialEventGroupsByNextTokenResult>
        <ResponseMetadata>
          <RequestId>1105b931-6f1c-4480-8e97-f3b46EXAMPLE</RequestId>
        </ResponseMetadata>
      </ListFinancialEventGroupsByNextTokenResponse>`
    )

    const res = parseListFinancialEventGroupsResponse(
      '/finances:ListFinancialEventGroupsByNextTokenResponse',
      doc,
      true
    )

    expect(res).toMatchSnapshot()
  })
})
