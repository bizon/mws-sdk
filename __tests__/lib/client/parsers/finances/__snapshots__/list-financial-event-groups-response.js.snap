// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.parsers.finances.list-financial-event-groups-response should parse the ListFinancialEventGroupsByNextTokenResponse example response from MWS doc 1`] = `
Object {
  "listFinancialEventGroupsResult": Object {
    "financialEventGroups": Array [
      Object {
        "accountTail": "1212",
        "beginningBalance": Object {
          "currencyAmount": 0,
          "currencyCode": "USD",
        },
        "convertedTotal": Object {
          "currencyAmount": 19,
          "currencyCode": "USD",
        },
        "financialEventGroupEnd": 2014-09-09T07:30:00.000Z,
        "financialEventGroupId": "22YgYW55IGNhcm5hbCBwbGVhEXAMPLE",
        "financialEventGroupStart": 2014-09-01T07:30:00.000Z,
        "fundTransferDate": 2014-09-09T07:30:00.000Z,
        "fundTransferStatus": "Successful",
        "originalTotal": Object {
          "currencyAmount": 19,
          "currencyCode": "USD",
        },
        "processingStatus": "Closed",
        "traceId": "128311029381HSADJEXAMPLE",
      },
    ],
    "nextToken": "2YgYW55IGNhcm5hbCBwbGVhcEXAMPLE",
  },
  "responseMetadata": Object {
    "requestId": "1105b931-6f1c-4480-8e97-f3b46EXAMPLE",
  },
}
`;

exports[`lib.client.parsers.finances.list-financial-event-groups-response should parse the ListFinancialEventGroupsResponse example response from MWS doc 1`] = `
Object {
  "listFinancialEventGroupsResult": Object {
    "financialEventGroups": Array [
      Object {
        "accountTail": "1212",
        "beginningBalance": Object {
          "currencyAmount": 0,
          "currencyCode": "USD",
        },
        "convertedTotal": Object {
          "currencyAmount": 19,
          "currencyCode": "USD",
        },
        "financialEventGroupEnd": 2014-09-09T07:30:00.000Z,
        "financialEventGroupId": "22YgYW55IGNhcm5hbCBwbGVhEXAMPLE",
        "financialEventGroupStart": 2014-09-01T07:30:00.000Z,
        "fundTransferDate": 2014-09-09T07:30:00.000Z,
        "fundTransferStatus": "Successful",
        "originalTotal": Object {
          "currencyAmount": 19,
          "currencyCode": "USD",
        },
        "processingStatus": "Closed",
        "traceId": "128311029381HSADJEXAMPLE",
      },
    ],
    "nextToken": "2YgYW55IGNhcm5hbCBwbGVhcEXAMPLE",
  },
  "responseMetadata": Object {
    "requestId": "1105b931-6f1c-4480-8e97-f3b46EXAMPLE",
  },
}
`;
